cmd/start.go

```diff
@@ -0,0 +1,60 @@
+package cmd
+
+import (
+	"os/signal"
+	"syscall"
+
+	"github.com/spf13/cobra"
+
+	"github.com/celestiaorg/celestia-node/node"
+)
+
+// Start constructs a CLI command to start Celestia Node daemon of the given type 'tp'.
+// It is meant to be used a subcommand and also receive persistent flag name for repository path.
+func Start(repoName string, tp node.Type) *cobra.Command {
+	if !tp.IsValid() {
+		panic("cmd: Start: invalid Node Type")
+	}
+	if len(repoName) == 0 {
+		panic("parent command must specify a persistent flag name for repository path")
+	}
+	return &cobra.Command{
+		Use:          "start",
+		Short:        "Starts Node daemon. First stopping signal gracefully stops the Node and second terminates it.",
+		Aliases:      []string{"run", "daemon"},
+		Args:         cobra.NoArgs,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			repoPath := cmd.Flag(repoName).Value.String()
+
+			repo, err := node.Open(repoPath, tp)
```

Comment:
This is confusing due to the package name. Suggest you just call this from inside `New()` rather than calling the "repo" `Open()` and then feeding that to `New()`

Reply:
@mattd, ‘New’ depends on Repository for a reason and you can't just hide opening inside it. The reason is that we want to support different Repository implementation, specifically in-memory ones for testing.

Reply:
However, I agree that ‘node.Open’ might be confusing. To resolve this, I propose renaming it to ‘node.OpenRepo(sitory)’.

Reply:
Why did you call it "Repo" and not something like "Store"? I think that might be more standard in this context.

Reply:
This is also out of scope and should have been raised in several previous PR’s touching Repository.

Reply:
To answer your question. I don't mind about the name tbh. I just see the pattern of having something to manage all on-disk data useful for our case. 

The name Repository is taken from the Protocol Labs ecosystem, as their projects and multiple projects from other connected teams use it in this case. 

The “Store” is indeed a widely used term in similar contexts. But I see a Store being attached to some specific type, e.g. KVStore or Keystore, while a Repository is a store for all stores. This is how I see it and made it work, but we can change the semantics here if needed.

Reply:
Protocol labs is wrong too then lol. Store is definitely more general than "keystore", it's often used for "memstore" or "datastore" and many other things.

It's not critical to change, but I think that as you suggested you should at least change the "node.Open" to "node.OpenRepo" to make it less confusing. Or perhaps even more the "repo" interface into another package entirely?

Reply:
Suggestion to rename to `OpenRepo` SGTM

Reply:
we'll do it in a separate PR later https://github.com/celestiaorg/celestia-node/issues/126